core/src/conn.rs

```diff
@@ -234,8 +234,9 @@ impl MongoConnection {
                 // Ensure the library is loaded if Enterprise edition is detected
                 if get_mongosqltranslate_library().is_none() {
                     return Err(Error::UnsupportedClusterConfiguration(
-                        "Enterprise edition detected, but mongosqltranslate library not found."
-                            .to_string(),
+                        "Enterprise edition detected, but mongosqltranslate library not found. \
+                        You are either (1) trying to connect to an on-premises cluster and have the wrong driver \
+                        or are (2) trying to connect to an Atlas cluster directly instead of through an ADF instance.".to_string(),
```

Comment:
This message is the only difference between what's in this PR and what's in the on-prem driver. I think this will help prevent future confusion.